seisbench/data/inspection.py

```diff
@@ -0,0 +1,410 @@
+            .groupby(pd.Grouper(key="source_origin_time", freq="D"))
+        )
+        events = []
+        stations = set()
+
+        all_picks = []
+        with tqdm(desc="Dumping MiniSEEDs", total=n_events) as pbar:
+            for day, metadata_grouped in day_groups:
+                n_events_day = metadata_grouped.shape[0]
+                if n_events_day == 0:
+                    continue
+
+                pick_markers = []
+                traces = []
+                for event_idx in range(n_events_day):
+                    event_traces, event, event_picks, stas = self._get_pyrocko_data(
+                        metadata_grouped.iloc[event_idx]["source_id"]
+                    )
+                    traces += event_traces
+                    pick_markers += event_picks
+                    events.append(event)
+                    stations.update(stas)
+
+                filename = directory / "mseed" / f"{day.date()}.mseed"
+                filename_picks = directory / "picks" / f"{day.date()}.picks"
+
+                try:
+                    traces = degapper(sorted(traces, key=lambda tr: tr.full_id))
+                except NoData:
+                    logger.error("Skipping day %s due to degapper NoData error", day)
+                    continue
+                save(traces, str(filename), format="mseed", overwrite=True)
+
+                save_markers(pick_markers, str(filename_picks))
+                all_picks += pick_markers
+
+                seisbench.logger.info("Wrote %d events to %s", n_events_day, filename)
+                pbar.update(n_events_day)
+
+        seisbench.logger.info("Dumping events, picks and stations")
+        pyrocko_stations = [sta.as_pyrocko_station() for sta in stations]
+
+        dump_events(events, str(directory / "events.yaml"), format="yaml")
+        save_markers(all_picks, str(directory / "all_picks.picks"))
+        dump_stations_yaml(pyrocko_stations, str(directory / "stations.yaml"))
+
+        dump_events_csv(events, directory / "csv" / "events.csv")
+        dump_stations_csv(list(stations), directory / "csv" / "stations.csv")
+
+        readme = directory / "README.md"
+        readme.write_text(
+            f"""
+# SeisBench MiniSEED dump of dataset {ds.name}
+
+This directory contains a MiniSEED dump of the SeisBench dataset '{ds.name}'.
+The dataset contains {len(ds)} earthquake records associated with {n_events} events.
+
+The data is organized as follows:
+- `mseed/`: Contains MiniSEED files, one file per day.
+- `picks/`: Contains pick files in pyrocko .picks format, one file per day.
+- `events.yaml`: Contains all events in pyrocko .yaml format.
+- `stations.yaml`: Contains all stations in pyrocko .yaml format.
+- `all_picks.picks`: Contains all picks in pyrocko .picks format
+
+## Open Pyrocko Snuffler
+
+You can open the data in pyrocko's snuffler using the following command:
+```bash
+squirrel snuffler -a mseed/*.mseed -e events.yaml -s stations.yaml
```

Comment:
Do the picks need to be loaded explicitly as well or is that part of loading the events?